taas/mapping.py

```diff
@@ -0,0 +1,99 @@
+from abc import ABCMeta, abstractmethod
+
+
+class Mapping(object):
+    """
+        This class handles mappings between our raw input, and what we generate.
```

Comment:
1 recommendation on this point, choose either Google or NumPy notation for docstrings, and use it everywhere. Boy do I wish I'd started doing this sooner. From Sphinx (python documentation generator) https://sphinxcontrib-napoleon.readthedocs.io/en/latest/